### PR TITLE
Added page for writing hexpansion apps

### PR DESCRIPTION
**Pull request by walkerdanny**
New page specifically for hexpansion apps - includes three different examples to find what slot the hexpansion is in (two of which are tested).

Includes a bit of information about the HexpansionConfig class and how to interact with it.

### CODE REVIEW

**Review by npentrel — APPROVED** [review]
Thank you - this is great!
Will merge but have one comment and one question 😊

File: docs/hexpansions/writing-hexpansion-apps.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            self.hexpansion_config = self.scan_for_hexpansion()
+
+            eventbus.on(HexpansionInsertionEvent, self.handle_hexpansion_insertion, self)
+            eventbus.on(HexpansionRemovalEvent, self.handle_hexpansion_removal, self)
+
+        def handle_hexpansion_insertion(self, event):
+            self.hexpansion_config = self.scan_for_hexpansion()
+
+
+        def handle_hexpansion_removal(self, event):
```
[the remark this answers is not in the document]
Comment (walkerdanny) [reply docs/hexpansions/writing-hexpansion-apps.md:96]: Good to know! I never bother poking these things. I'll add that in when I find the time to expand the HexpansionInsertion/Removal event documentation.

File: docs/hexpansions/writing-hexpansion-apps.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+## The HexpansionConfig class
+
+The `HexpansionConfig` object that you get after following the examples is where the magic all happens. It allows you to access the following:
+
+| Object | Description | Example Usage |
+| ------ | ----------- | --------- |
+| `HexpansionConfig.port` | The port number your hexpansion is connected to. | None |
+| `HexpansionConfig.pin[]` | A list of 4 `Pin` objects, which are the high-speed GPIO pins for this hexpansion slot. | [See MicroPython Docs](https://docs.micropython.org/en/latest/library/machine.Pin.html)  | 
+| `HexpansionConfig.ls_pin[]` | A list of 4 `ePin` objects for this hexpansion slot. |  [See eGPIO](../tildagon-apps/reference/badge-hardware.md#egpio) |
```
[the remark this answers is not in the document]
Comment (walkerdanny) [reply docs/hexpansions/writing-hexpansion-apps.md:212]: Do you mean the difference between the Pin and ePin objects? I understand the differences but maybe not all of the subtleties.

[the remark this answers is not in the document]
Comment (walkerdanny) [reply docs/hexpansions/writing-hexpansion-apps.md:212]: Oops, just noticed there's a typo here too - there's 5 of the eGPIO in a port, not 4.

[the remark this answers is not in the document]
Comment (npentrel) [reply docs/hexpansions/writing-hexpansion-apps.md:212]: Yeah wondering if we could add a sentence to explain what each is for or something. I don't know whether that's generally known. I don't know, I've not been able to figure it out with Google.

[the remark this answers is not in the document]
Comment (npentrel) [reply docs/hexpansions/writing-hexpansion-apps.md:212]: I'll see if I can poke the team to explain

**Pull request merged**